pkg/controller/csi_handler_test.go

```diff
@@ -1381,19 +1395,33 @@ func TestCSIHandler(t *testing.T) {
 
 func TestCSIHandlerReconcileVA(t *testing.T) {
 	nID := map[string]string{
-		vaNodeIDAnnotation: testNodeName,
+		vaNodeIDAnnotation: testNodeID,
+	}
+	vaGroupResourceVersion := schema.GroupVersionResource{
+		Group:    storage.GroupName,
+		Version:  "v1",
+		Resource: "volumeattachments",
 	}
-
 	tests := []testCase{
 		{
 			name: "va attached actual state not attached",
 			initialObjects: []runtime.Object{
 				va(true /*attached*/, "" /*finalizer*/, nID /*annotations*/),
 				pvWithFinalizer(),
+				csiNode(),
 			},
 			listerResponse: map[string][]string{
 				// Intentionally empty
 			},
+			expectedActions: []core.Action{
+				// Finalizer is saved first
+				core.NewPatchAction(vaGroupResourceVersion, metav1.NamespaceNone, testPVName+"-"+testNodeName,
+					types.MergePatchType, patch(va(false /*attached*/, "" /*finalizer*/, nil /* annotations */),
+						va(false /*attached*/, fin, nil))),
+				core.NewPatchSubresourceAction(vaGroupResourceVersion, metav1.NamespaceNone, testPVName+"-"+testNodeName,
+					types.MergePatchType, patch(va(false /*attached*/, "", nil),
+						va(false, "", nil)), "status"),
```

Comment:
This change is odd:

* The patch is empty.  My tests show that the attacher indeed executes empty PATCH `{}`. Is there a bug in the attacher?

* While the resulting patch is empty, the way it's constructed is wrong - it expects that the volume gets detached (`va(false, ...`) - should it be `true`? The volume was attached successfully.

Reply:
Haven't found if there is a bug executing empty patch.
Fixed the patch construction logic in the test.